tools/build-host-image.sh

```diff
@@ -117,10 +119,41 @@ function main
 
     pfexec zfs create -p rpool/images/$USER
 
+    HELIOS_REPO=https://pkg.oxide.computer/helios-netdev
+
+    # We have to do a bit of extra work to find the host OS hash
+    OS_HASH="unknown"
+
+    tmpd=$(mktemp -d)
+    if [[ -d "$tmpd" ]]; then
+        pkgrecv -s "$HELIOS_REPO" -d "$tmpd" --raw -m latest SUNWcs
+        if (($? == 0)); then
+            for PKGDIR in "$tmpd/SUNWcs/"*; do
+                if [[ -d "$PKGDIR" ]]; then
+                    BUILDHASH=$( \
+                        grep 'path=etc/versions/build ' \
+                        "$PKGDIR/manifest.file" | \
+                        awk '{print $2}' \
+                    )
+                    if [[ -n "$BUILDHASH" && -f "$PKGDIR/$BUILDHASH" ]]; then
+                        OS_HASH="$(awk -F- '{print $NF}' "$PKGDIR/$BUILDHASH")"
+                        break
+                    fi
+                fi
+            done
+        fi
+        rm -rf "$tmpd"
+    fi
```

Comment:
I feel like this could potentially be misleading if new packages arrive between when we do this generation, and when we then do the work below to actually install things; i.e., the hash might be wrong depending on things that are occurring asynchronously outside this process.

Could we instead perhaps add support to `helios-build` for macro expansion in the `-N` argument?  You can borrow the expansion stuff from https://github.com/illumos/image-builder/blob/main/src/expand.rs as I think it would likely work well there also.  (I wrote that while at Oxide)

Then we could say something like:

```
-N "ci-${GITHUB_SHA:0:7}-\${os_short_commit}"
```

and have `${os_short_commit}` expanded to basically what you're getting from `${OS_HASH:1:8}` there by `helios-build` itself, which can inspect the ramdisk root to find whatever it needs.

Reply:
That's a good idea to solve the chicken-and-egg problem, thanks, I will look into that.